examples/ReferenceTest/reference_client.py

```diff
@@ -0,0 +1,175 @@
+import time
+import traceback
+import os
+import sdc11073
+from collections import defaultdict
+from sdc11073 import observableproperties
+from sdc11073.definitions_sdc import SDC_v1_Definitions
+from concurrent import futures
+from sdc11073.certloader import mk_ssl_context
+
+
+adapter_ip = os.getenv('ref_ip') or '127.0.0.1'
+ca_folder = os.getenv('ref_ca')  # or None
+search_epr = os.getenv('ref_search_epr')  or 'abc' # abc is fixed ending in reference_device uuid.
+
+def run_ref_test():
+    results = []
+    print('Test step 1: discover device which endpoint ends with "{}"'.format(search_epr))
+    wsd = sdc11073.wsdiscovery.WSDiscoveryWhitelist([adapter_ip])
+    wsd.start()
+    my_service = None
+    while my_service is None:
+        services = wsd.searchServices(types=SDC_v1_Definitions.MedicalDeviceTypesFilter)
+        print('found {} services {}'.format(len(services), ', '.join([s.getEPR() for s in services])))
+        for s in services:
+            if s.getEPR().endswith(search_epr):
+                my_service = s
+                print('found service {}'.format(s.getEPR()))
+                break
+    print('Test step 1 successful: device discovered')
+    results.append('### Test 1 ### passed')
+
+    print('Test step 2: connect to device...')
+    try:
+        if ca_folder:
+            ssl_context = mk_ssl_context(ca_folder, "client")

```

Comment:
I am not a fan of the naming scheme used for certificate files in mk_ssl_context, but that's an issue separate from this pull request